posts/2021-03-25-Rust-1.51.0.md

```diff
@@ -0,0 +1,174 @@
+---
+layout: post
+title: "Announcing Rust 1.51.0"
+author: The Rust Release Team
+release: true
+---
+
+The Rust team is happy to announce a new version of Rust, 1.51.0. Rust is a
+programming language that is empowering everyone to build reliable and
+efficient software.
+
+If you have a previous version of Rust installed via rustup, getting Rust
+1.51.0 is as easy as:
+
+```console
+rustup update stable
+```
+
+If you don't have it already, you can [get `rustup`][install]
+from the appropriate page on our website, and check out the
+[detailed release notes for 1.51.0][notes] on GitHub.
+
+[install]: https://www.rust-lang.org/install.html
+[notes]: https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1510-2021-03-25
+
+## What's in 1.51.0 stable
+This release represents one of the largest additions to the Rust language and cargo in quite a while. With the stabilization of const generics, and the new feature resolver, along with more minor additions such as the `split-debuginfo` option, and the `addr_of!` API. Let's dive right into it!
+
+
+### Const Generics MVP
+Before this release, Rust allowed you to haves your types be parameterized over lifetimes or types. For example if we wanted to have a `struct` that is generic over an array of 32 items, we'd write the following:
+
+```rust
+struct FixedArray<T> {
+              // ^^^ Type generic definition
+    list: [T; 32]
+        // ^ Where we're using it.
+}
+
+// `FixedArray<u8>` becomes
+struct FixedArray_u8 {
+    list: [u8; 32]
+}
+```
```

Comment:
I think this should be two separate code blocks (and the same for the code block a little later on) because the user *doesn't* write the monomorphized version, but you say "we'd write the following". Also I think it looks nicer as two code blocks.

`````suggestion
```rust
struct FixedArray<T> {
              // ^^^ Type generic definition
    list: [T; 32]
        // ^ Where we're using it.
}
```

If we then use `FixedArray<u8>`, the compiler will make a monomorphic version of `FixedArray` that looks like:

```
struct FixedArray_u8 {
    list: [u8; 32]
}
```
`````